config/initializers/es.rb

```diff
@@ -20,20 +21,15 @@ def self.client
     )
   end
 
-  def self.create_index
-    index_name = ENV.fetch('SEARCHELASTIC_INDEX')
-
-    template_generator = SearchElastic::Template.new("*#{index_name}*")
-
-    self.client.indices.put_template(
-      body: template_generator.body,
-      create: true,
-      include_type_name: false,
-      name: index_name,
-      order: 0
-    )
+  @index_creator = SearchElasticIndexCreator.new(
```

Comment:
What do you think about setting up the client directly when creating this object, rather when calling the `create_or_update_index` method?